src/standalone/env.go

```diff
@@ -79,12 +76,33 @@ func (env *environment) setRequiredFields() error {
 }
 
 func (env *environment) setOptionalFields() {
```

Comment:
should be split into separate functions, e.g. injection/base and optional fields